components/service/telemetry/src/main/java/org/mozilla/telemetry/Telemetry.java

```diff
@@ -35,6 +35,10 @@
 import kotlin.Unit;
 import kotlin.jvm.functions.Function0;
 
+/**
+ * @deprecated The whole service-telemetry library is deprecated. Please use the Glean SDK instead.
+ */
+@Deprecated
```

Comment:
nit: Some tests may break since they are using a deprecated API now (We may need to propagate the annotation).

Reply:
Do you mean application tests? Nothing seems to be breaking on ci (luckily :D )

Reply:
Okay, then ignore! :)

Reply:
Ah, you are probably lucky because this is Java. Those checks are only enabled for Kotlin. :)